container/starbuck.py

```diff
@@ -0,0 +1,187 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import logging
+import os
+import re
+
+import ruamel.yaml
+from ruamel.yaml.comments import CommentedMap
+
+from .utils import create_role_from_templates
+from .exceptions import AnsibleContainerException
+
+logger = logging.getLogger(__name__)
+
+
+class Starbuck(object):
+    base_path = None
+    alternate_file_name = None
+    docker_file_path = None
+    dockerfile = {}
+    cached_instructions = None
+    project_name = None
+    role_name = None
+    role_path = None
+
+    def __init__(self, base_path, project_name, alternate_file_name):
+        self.base_path = base_path
+        self.project_name = project_name
+        self.alternate_file_name = alternate_file_name
+        file_name = self.alternate_file_name if self.alternate_file_name else 'Dockerfile'
+        self.docker_file_path = os.path.normpath(os.path.join(self.base_path, file_name))
+        self.role_name = "dockerfile-to-ansible"
+        self.role_path = os.path.normpath(os.path.join(self.base_path, 'roles', self.role_name))
+
+        #ruamel.yaml.representer.RoundTripRepresenter.add_representer(ordereddict,
+        #                                                             RoundTripRepresenter.represent_ordereddict)
+
+    def assert_dockerfile_exists(self):
+        if not os.path.exists(self.docker_file_path):
+            raise AnsibleContainerException("Failed to find {}".format(self.docker_file_path))
+
+    @property
+    def lines(self):
+        byte_to_string = lambda x: x.decode('utf-8') if isinstance(x, bytes) else x
+        with open(self.docker_file_path, 'r') as f:
+            return [byte_to_string(line) for line in f.readlines()]
+
+    @property
+    def instructions(self):
+        '''
+        Create a list of Dockerfile instructions from self.lines.
+        :return: list
+        '''
+
+        if self.cached_instructions:
+            return self.cached_instructions
+
+        strip_line = lambda x: re.sub(r'\\$', '', x.rstrip())
```

Comment:
If you're going to the trouble of decoding the Dockerfile from UTF-8, you should use unicode strings here too.